server/task/task-distributed/src/main/java/org/apache/james/task/eventsourcing/distributed/TasksSerializationModule.java

```diff
@@ -20,71 +20,78 @@
 package org.apache.james.task.eventsourcing.distributed;
 
 import java.util.List;
-import java.util.function.Function;
+import java.util.function.BiFunction;
 import java.util.stream.Stream;
 
 import org.apache.james.eventsourcing.eventstore.cassandra.dto.EventDTOModule;
+import org.apache.james.server.task.json.JsonTaskAdditionalInformationsSerializer;
 import org.apache.james.server.task.json.JsonTaskSerializer;
 import org.apache.james.task.eventsourcing.CancelRequested;
 import org.apache.james.task.eventsourcing.Cancelled;
 import org.apache.james.task.eventsourcing.Completed;
 import org.apache.james.task.eventsourcing.Created;
 import org.apache.james.task.eventsourcing.Failed;
 import org.apache.james.task.eventsourcing.Started;
+import org.apache.james.task.eventsourcing.TaskEvent;
 
 import com.github.steveash.guavate.Guavate;
 
 public interface TasksSerializationModule {
-    Function<JsonTaskSerializer, EventDTOModule<Created, CreatedDTO>> CREATED = jsonTaskSerializer -> EventDTOModule
+    @FunctionalInterface
+    interface TaskSerializationModuleFactory<T extends TaskEvent, U extends TaskEventDTO> {
+        EventDTOModule<T, U> create(JsonTaskSerializer taskSerializer, JsonTaskAdditionalInformationsSerializer additionalInformationsSerializer);
+    }
+
+    TaskSerializationModuleFactory<Created, CreatedDTO> CREATED = (jsonTaskSerializer, jsonTaskAdditionalInformationsSerializer) -> EventDTOModule
```

Comment:
well, I know I asked for these types but after working a bit with this code I can't understand why it's not regular static methods.

Reply:
It's even more verbose with static methods, we'll keep it that way